spec/support/sign_in/certifiable_shared_examples.rb

```diff
@@ -0,0 +1,183 @@
+# frozen_string_literal: true
+
+RSpec::Matchers.define :matching_public_key_for do |expected_public_key|
+  match do |actual_public_key|
+    actual_public_key.to_der == expected_public_key.to_der
+  end
+end
+
+RSpec.shared_examples 'implements certifiable concern' do
```

Comment:
I think you'll have to TImecop.freeze to a specific time for the expired/expiring/valid related specs. Otherwise the behavior will change in a couple months from now

Reply:
Good call, señor!

Reply:
The fixture I'm using for the expired cert is already expired, but I added the timecop for the expiring cert specs.